src/ImageSharp/Common/Helpers/Vector128Utilities.cs

```diff
@@ -25,43 +23,56 @@ internal static class Vector128_
 #pragma warning restore SA1649 // File name should match first type name
 {
     /// <summary>
-    /// Gets a value indicating whether shuffle operations are supported.
+    /// Average packed unsigned 8-bit integers in <paramref name="left"/> and <paramref name="right"/>, and store the results.
     /// </summary>
-    public static bool SupportsShuffleNativeByte
+    /// <param name="left">
+    /// The first vector containing packed unsigned 8-bit integers to average.
+    /// </param>
+    /// <param name="right">
+    /// The second vector containing packed unsigned 8-bit integers to average.
+    /// </param>
+    /// <returns>
+    /// A vector containing the average of the packed unsigned 8-bit integers
+    /// </returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static Vector128<byte> Average(Vector128<byte> left, Vector128<byte> right)
     {
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        get
+        if (Sse2.IsSupported)
         {
-            if (Vector128.IsHardwareAccelerated)
-            {
-                if (RuntimeInformation.ProcessArchitecture is Architecture.X86 or Architecture.X64)
-                {
-                    return Ssse3.IsSupported;
-                }
-
-                return true;
-            }
+            return Sse2.Average(left, right);
+        }
 
-            return false;
+        if (AdvSimd.IsSupported)
+        {
+            return AdvSimd.FusedAddRoundedHalving(left, right);
         }
-    }
 
-    /// <summary>
-    /// Gets a value indicating whether right align operations are supported.
-    /// </summary>
-    public static bool SupportsAlignRight
-    {
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        get => Ssse3.IsSupported || AdvSimd.IsSupported;
+        // Portable fallback: (a + b + 1) >> 1
+        return (left + right + Vector128.Create((byte)1)) >> 1;
```

Comment:
This isn't quite the same.

`Sse2.Average` (pavg) and `AdvSimd.FusedAddRoundedHalving` (urhadd) are instead effectively doing:
```csharp
return Vector128.Narrow(
    (Vector128.WidenLower(left) + Vector128.WidenLower(right) + Vector128<ushort>.One) >> 1,
    (Vector128.WidenUpper(left) + Vector128.WidenUpper(right) + Vector128<ushort>.One) >> 1
);
```

Which is to say that, it accounts for a potential 9th bit to ensure a correct truncated result. This can be observed for `Average(255, 1)` which should produce `128` but where this fallback will produce `0`